### PR TITLE
Convert is_volume, is_volume_dir, and is_volume_file for PDS4 (fix #6…

### LINKED ISSUE #6: Unknown opus_products for HSTU0_8634/DATA/VISIT_07/U62M0702R.png
2019-06-10 11:30:15 | pds.opus_import.main |--| WARNING | [HSTU0_8634 index row 24] Empty opus_product key for files: /Volumes/pdsdata-server2/holdings/volumes/HSTUx_xxxx/HSTU0_8634/DATA/VISIT_07/U62M0702R.png

This is the only HST preview image that's a PNG file instead of a JPG.

### LINKED ISSUE #69: Accommodation for pds4file, update the code to make sure `exists` and `logical_path` work
- Modify the code the make sure these two properties  (frequently used) `exists` and `logical_path` work.

### PR DESCRIPTION
**Pull request by mace-space**
…4). Should work for bundle sets (as well as bundles). Also, code update resolves #69, #65, #62.

### CODE REVIEW

**Comment by juzen2003** [comment]
After discussion in the meeting, let's restore the original functionality to return '' as the falsey value. And change the function names:
is_volume_* => is_bundle_*
is_volset_* => is_bundleset_*

File: Pds4File/pds4file.py [hunk 1]
```diff
@@ -2968,37 +2968,37 @@ def volset_pdsfile(self, category=None, rank=None):
...
-        return (self.volname and not self.volname_ or False) # MJTM: 'or False' account for bundle sets
+    def is_bundle_file(self):
+        """True if this is a bundle-level checksum or archive file."""
+        return (self.bundlename and not self.bundlename_) # Note that a bundle set will return an empty string '' rather than False
+        #return (self.volname and not self.volname_ or False) # MJTM: 'or False' account for bundle sets
 
     @property
-    def is_volume(self):
+    def is_bundle(self):
         """True if this is a volume-level file, be it a directory or a
```
Comment (juzen2003) [inline Pds4File/pds4file.py:2984]: Update the function comments for these functions:
```
is_bundle
is_bundleset_dir
is_bundleset_file
```

**Review by juzen2003 — CHANGES_REQUESTED** [review]
We want to rename `volume*` to `bundle*` and `volset*` to `bundleset*`, so this means we will also rename `self.volset*` to `self.bundleset*` and `self.volname*` to `self.bundlename*`

If we search `pds4file.py`, there are still 33 `self.volset*` and 29 `self.volname*` in the code, please update them as well to avoid error. One way to test out if the new changes break anything is to run simple import and instantiate a pdsfile instance. By doing so, we can see an error in the setup in ipython, and we know there are more changes needed to be done to fix the error. 
```
In [1]: import pds4file
   ...: pds4file.use_shelves_only(False)
   ...: pds4file.preload('/Users/yu-jenchang/Dropbox (SETI Institute)/Pds4FileTest/pds4-holdings/bundles')
   ...: 

In [2]: a = pds4file.PdsFile.from_abspath('/Users/yu-jenchang/Dropbox (SETI Institute)/Pds4FileTest/pds4-h
   ...: oldings/bundles/uranus_occs_earthbased/uranus_occ_u0_kao_91cm/data/atmosphere/u0_kao_91cm_734nm_co
   ...: unts-v-time_atmos_ingress.xml')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [2], in <cell line: 1>()
----> 1 a = pds4file.PdsFile.from_abspath('/Users/yu-jenchang/Dropbox (SETI Institute)/Pds4FileTest/pds4-holdings/bundles/uranus_occs_earthbased/uranus_occ_u0_kao_91cm/data/atmosphere/u0_kao_91cm_734nm_counts-v-time_atmos_ingress.xml')

File ~/Desktop/seti/workspace/pds-webtools/Pds4File/pds4file.py:3544, in PdsFile.from_abspath(abspath, fix_case, must_exist, caching, lifetime)
   3542 # Handle the rest of the tree using child()
   3543 for part in parts[pds4_holdings_index + 1:]:
-> 3544     this = this.child(part, fix_case=fix_case, must_exist=must_exist,
   3545                             caching=caching, lifetime=lifetime)
   3547 if must_exist and not this.exists:
   3548     raise IOError('File not found', this.abspath)

File ~/Desktop/seti/workspace/pds-webtools/Pds4File/pds4file.py:3364, in PdsFile.child(self, basename, fix_case, must_exist, caching, lifetime, allow_index_row)
   3360         if this.voltype_[:-1] not in VOLTYPES:
   3361             raise ValueError('Unrecognized volume type "%s": %s' %
   3362                              (this.voltype_[:-1], this.logical_path))
-> 3364         return this._complete(must_exist, caching, lifetime)
   3366     raise ValueError('Cannot define child from PDS root: ' +
   3367                      this.logical_path)
   3369 ### Resume caching no matter what
   3370 finally:

File ~/Desktop/seti/workspace/pds-webtools/Pds4File/pds4file.py:3129, in PdsFile._complete(self, must_exist, caching, lifetime)
   3126     if '/' in self.logical_path:
   3127         CACHE.set(self.logical_path.lower(), self, lifetime=lifetime)
-> 3129     self._update_ranks_and_vols()
   3131 return self

File ~/Desktop/seti/workspace/pds-webtools/Pds4File/pds4file.py:3148, in PdsFile._update_ranks_and_vols(self)
   3145 if not LOCAL_PRELOADED:     # we don't track ranks without a preload
   3146     return
-> 3148 if self.volset and not self.volname:
   3149     key = self.volset
   3150 elif self.volname and not self.volname_:

AttributeError: 'PdsFile' object has no attribute 'volset'
```

Note: as we discussed in the meeting, this is the case when adding some simple tests come in handy to catch errors like this.

File: Pds4File/pds4file.py (hunk 1 above)
Comment (mace-space) [reply Pds4File/pds4file.py:2984]: Thanks, I hadn't been using the preload function previously so there weren't any errors reported:

```
In [1]: import pds4file

In [2]: a = pds4file.PdsFile.from_abspath('/Users/mm9684/DropboxSETI/Dropbox (SETI Institute)/Pds4FileTest/pds4-holdings/bundles/uranus_occs_ea
   ...: rthbased/uranus_occ_u0_kao_91cm/data/atmosphere/u0_kao_91cm_734nm_counts-v-time_atmos_ingress.xml')

In [3]: a.exists
Out[3]: True

In [4]: a.bundlename
Out[4]: 'uranus_occ_u0_kao_91cm'

In [5]: a.bundleset
Out[5]: 'uranus_occs_earthbased'

In [6]: a.is_bundleset_file
Out[6]: False

In [7]: a.is_bundle_file
Out[7]: False
```
But when I do use preload() I reproduce the same error, which I will fix by updating all instances of `self.volname*` and `self.volset*` to their "bundle" equivalents

**Review by juzen2003 — APPROVED** [review]

**Pull request merged**